table_schema_cache.go

```diff
@@ -293,6 +293,31 @@ func (c TableSchemaCache) Get(database, table string) *TableSchema {
 	return c[fullTableName(database, table)]
 }
 
+// Helper to sort a given map of tables with a second list giving a priority.
+// If an element is present in the input and the priority lists, the item will
+// appear first (in the order of the priority list), all other items appear in
+// the order given in the input
+func (c TableSchemaCache) GetTableListWithPriority(priorityList []string) (tableNames []string) {
+	// just a fast lookup if the list contains items already
+	contains := map[string]bool{}
```

Comment:
You can do `map[sting]struct{}`, as that's the standard way to express a set in Golang.